backend/lib/endpoints/posts.js

```diff
@@ -366,6 +374,16 @@ async function routes(app) {
         throw app.httpErrors.notFound();
       }
 
+      if (actor) {
+        // user shouldn't see reported posts on post page
+        const didReport = post.reportedBy
+          ? post.reportedBy.find(
+              (r) => r.id.toString() === actor._id.toString(),
+            )
+          : false;
+        if (didReport) throw app.httpErrors.notFound();
```

Comment:
should we throw `notFound` or send back empty result here? I imagine when user query post he/she reports, we don't report error? let me know if I didn't get the scenario here

Reply:
This will show the normal "post was deleted or expired", it works as expected.
we don't return empty result because this is looking for a single post by id, /posts/:postId
So you return 404, return an empty array if it was supported to get multiple posts.